### PR TITLE
Add ?link command

### PR DESCRIPTION
**Pull request by Kacarott**
Provides a quick way to access useful links, usually when guiding a user to a particular page.

Example:
```
no0bUser99: Hey guys I want to create my first kata, how do I do this?
Kacarott:   You should read this:
Kacarott:   ?link authoring
Codewars:   https://docs.codewars.com/authoring
no0bUser99: Ok thanks!
```

Currently implemented `?link`s:
- [docs](https://docs.codewars.com/)
- [honor](https://docs.codewars.com/gamification/honor)
- [katas](https://www.codewars.com/kata/latest/my-languages)
- [authoring](https://docs.codewars.com/authoring)
- [ranking](https://docs.codewars.com/curation/kata)
- [troubleshooting](https://docs.codewars.com/training/troubleshooting)
- [github](https://github.com/codewars/codewars.com/issues)

### CODE REVIEW

File: src/events/message/commands/help/link.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // Input validation
+  const result = z.tuple([word((w) => LINKS.hasOwnProperty(w))]).safeParse(args);
+  if (!result.success) {
+    message.reply(USAGE);
+    return;
+  }
+  const [link] = result.data;
+
+  // Respond with link
+  await message.channel.send(`<${LINKS[link]}>`);
```
Comment (kazk) [inline src/events/message/commands/help/link.ts:28]: What do you think of replying with `See <url>` instead?
```
no0bUser99: Hey guys I want to create my first kata, how do I do this?
Kacarott:   ?link authoring
Codewars:   See https://docs.codewars.com/authoring
no0bUser99: Ok thanks!
```

File: src/events/message/commands/help/link.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// "?link <topic>" provides a simple shortcut to various useful URLs
+
+const LINKS: { [k: string]: string } = {
+  docs: "https://docs.codewars.com/",
+  honor: "https://docs.codewars.com/gamification/honor",
+  katas: "https://www.codewars.com/kata/latest/my-languages",
+  authoring: "https://docs.codewars.com/authoring",
+  ranking: "https://docs.codewars.com/curation/kata",
+  troubleshooting: "https://docs.codewars.com/training/troubleshooting",
+  github: "https://github.com/codewars/codewars.com/issues",
```
Comment (kazk) [inline src/events/message/commands/help/link.ts:13]: I might want `github` mapped to `https://github.com/codewars`, but we can adjust these later.

**Review by kazk — APPROVED** [review]

File: src/events/message/commands/help/link.ts (hunk 1 above)
Comment (Kacarott) [reply src/events/message/commands/help/link.ts:28]: Yeah that might be a bit nicer.

File: src/events/message/commands/help/link.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// "?link <topic>" provides a simple shortcut to various useful URLs
+
+const LINKS: { [k: string]: string } = {
+  docs: "https://docs.codewars.com/",
+  honor: "https://docs.codewars.com/gamification/honor",
+  katas: "https://www.codewars.com/kata/latest/my-languages",
+  authoring: "https://docs.codewars.com/authoring",
+  ranking: "https://docs.codewars.com/curation/kata",
+  troubleshooting: "https://docs.codewars.com/training/troubleshooting",
+  github: "https://github.com/codewars/",
```
Comment (kazk) [inline src/events/message/commands/help/link.ts:13]: If you want to link https://github.com/codewars/codewars.com/issues maybe add `issues`? 

Again, we can tweak this later too. I can merge this if you're ready.

Comment (Kacarott) [reply src/events/message/commands/help/link.ts:13]: I think for now I'm ready. We can add more as we notice the need arise.

**Pull request merged**